packages/validation/src/utils/functions.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * Helper function to determine if value is a valid number and is positive.
+ * @param num Any integer value
+ * @returns True if the value is a valid number and is positive, false otherwise.
+ */
+export const isPositiveNumber = (num: number) => {
+	if (Number.isNaN(num) === false && num >= 0) {
+		return true;
+	}
+	return false;
+};
```

Comment:
TS `number` covers floats too https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#the-primitives-string-number-and-boolean

I think this is technically correct because of how JS implements IEEE standards with 0 being positive by "default". In practical terms and based on this being a boolean check, I imagine it's use case may result in 0 causing unexpected results because it's a falsy js value.

Reply:
I see what you're saying
Essentially this should be changed to `if (Number.isNaN(num) === false && Number.isInterger(num) && num > 0) {`

Reply:
Updated usage of isPositiveNumber & parseInt: [Update usage of isPositiveNum function](https://github.com/Pan-Canadian-Genome-Library/daco/pull/241/commits/a75d415956b355ad15f56814b5fc0291f495de97)